merge.go

```diff
@@ -0,0 +1,183 @@
+package neo4j
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/cloudprivacylabs/lpg"
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+func compareNativeNode(n1, n2 *lpg.Node) bool {
+	if !n1.GetLabels().IsEqual(n2.GetLabels()) {
+		return false
+	}
+	// Expected properties must be a subset
+	propertiesOK := true
+	n2.ForEachProperty(func(k string, v interface{}) bool {
+		pv, ok := v.(*ls.PropertyValue)
+		if !ok {
+			return true
+		}
+		v2, ok := n1.GetProperty(k)
+		if !ok {
+			log.Printf("Error at %s: %v: Property does not exist", k, v)
+			propertiesOK = false
+			return false
+		}
+		pv2, ok := v2.(*ls.PropertyValue)
+		if !ok {
+			log.Printf("Error at %s: %v: Not property value", k, v)
+			propertiesOK = false
+			return false
+		}
+		if !pv2.IsEqual(pv) {
+			if strings.ToLower(pv2.AsString()) != strings.ToLower(pv.AsString()) {
+				log.Printf("Error at %s: Got %v, Expected %v: Values are not equal", k, pv, pv2)
+				propertiesOK = false
+				return false
+			}
+		}
+		return true
+	})
+	if !propertiesOK {
+		log.Printf("Properties not same")
+		return false
+	}
+	log.Printf("True\n")
+	return true
+}
+
+func compareNativeEdge(e1, e2 *lpg.Edge) bool {
+	return e1.GetLabel() == e2.GetLabel() && ls.IsPropertiesEqual(ls.PropertiesAsMap(e1), ls.PropertiesAsMap(e2))
+}
+
+func merge(memGraph, dbGraph *lpg.Graph, memGraphIds, dbGraphIds map[*lpg.Node]int64, memEdges, dbEdges map[*lpg.Edge]int64) (*lpg.Graph, []string, error) {
+	grph := lpg.NewGraph()
+	ops := make([]string, 0)
+
```

Comment:
Merge algorithm:
First create a list of all entities in the memgraph
Create a list of all entities in dbgraph
Then work on each entity root of the mempgraph using entityIds
Find corresponding db entity root from the db graph
If found, mergeEntity (new func)
If not found, create the entity